File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    
    while len(scoville) > 1 and scoville[0] < K:
        res = heapq.heappop(scoville) + (heapq.heappop(scoville) * 2)
        heapq.heappush(scoville, res)
        answer += 1
    return answer if scoville[0] >= K else -1

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for(int n : scoville) {
            pq.offer(n);
        }
        int answer = 0;
        while(pq.peek() < K) {
            if(pq.size() >= 2) {
                int p1 = pq.poll();
                int p2 = pq.poll();  
                int res = p1 + (p2 * 2);
                pq.offer(res);
                answer ++;
            }else {
                return -1;
            }
        }
        
        return answer;
    }
}